src/Framework/AspNetCoreAnalyzers/src/Analyzers/Mvc/MvcAnalyzer.cs

```diff
@@ -0,0 +1,152 @@
+                            // No HTTP method.
+                            break;
+                        case WellKnownType.Microsoft_AspNetCore_Mvc_HttpDeleteAttribute:
+                            httpMethodsBuilder.Add("DELETE");
+                            break;
+                        case WellKnownType.Microsoft_AspNetCore_Mvc_HttpGetAttribute:
+                            httpMethodsBuilder.Add("GET");
+                            break;
+                        case WellKnownType.Microsoft_AspNetCore_Mvc_HttpHeadAttribute:
+                            httpMethodsBuilder.Add("HEAD");
+                            break;
+                        case WellKnownType.Microsoft_AspNetCore_Mvc_HttpOptionsAttribute:
+                            httpMethodsBuilder.Add("OPTIONS");
+                            break;
+                        case WellKnownType.Microsoft_AspNetCore_Mvc_HttpPatchAttribute:
+                            httpMethodsBuilder.Add("PATCH");
+                            break;
+                        case WellKnownType.Microsoft_AspNetCore_Mvc_HttpPostAttribute:
+                            httpMethodsBuilder.Add("POST");
+                            break;
+                        case WellKnownType.Microsoft_AspNetCore_Mvc_HttpPutAttribute:
+                            httpMethodsBuilder.Add("PUT");
+                            break;
+                        default:
+                            throw new InvalidOperationException("Unexpected well known type:" + match);
+                    }
+                    actionRoutes.Add(new ActionRoute(methodSymbol, routeUsage, httpMethodsBuilder.ToImmutable()));
+                }
+            }
+        }
+    }
+
+    private static readonly WellKnownType[] RouteAttributeTypes = new[]
+    {
+        WellKnownType.Microsoft_AspNetCore_Mvc_RouteAttribute,
+        WellKnownType.Microsoft_AspNetCore_Mvc_HttpDeleteAttribute,
+        WellKnownType.Microsoft_AspNetCore_Mvc_HttpGetAttribute,
+        WellKnownType.Microsoft_AspNetCore_Mvc_HttpHeadAttribute,
+        WellKnownType.Microsoft_AspNetCore_Mvc_HttpOptionsAttribute,
+        WellKnownType.Microsoft_AspNetCore_Mvc_HttpPatchAttribute,
+        WellKnownType.Microsoft_AspNetCore_Mvc_HttpPostAttribute,
+        WellKnownType.Microsoft_AspNetCore_Mvc_HttpPutAttribute
+    };
+
+    private static RouteUsageModel? GetRouteUsageModel(AttributeData attribute, RouteUsageCache routeUsageCache, CancellationToken cancellationToken)
+    {
+        if (attribute.ConstructorArguments.IsEmpty || attribute.ApplicationSyntaxReference is null)
+        {
+            return null;
+        }
+
+        if (attribute.ApplicationSyntaxReference.GetSyntax(cancellationToken) is AttributeSyntax attributeSyntax &&
+            attributeSyntax.ArgumentList is { } argumentList)
+        {
+            var attributeArgument = argumentList.Arguments[0];
+            if (attributeArgument.Expression is LiteralExpressionSyntax literalExpression)
```

Comment:
Why go back to syntax? I guess `attribute.ConstructorArguments` can be used.

Reply:
routeUsageCache.Get (below) takes a token